docs/how-tos/build-a-plugin.md

```diff
@@ -0,0 +1,192 @@
+
+Don't worry, the following how-to will guide you how to write a reader your own data.
+
+
+## Getting started
+You should start by creating a clean repository that implements the following structure (for a plugin called ```pynxtools-plugin```):
+```
+pynxtools-plugin
+├── .github/workflows
+├── docs
+│   ├── explanation
+│   ├── how-tos
+│   ├── reference
+│   ├── tutorial
+├── src
+│   ├── pynxtools_plugin
+│       ├── reader.py
+├── tests
+│   └── data
+├── LICENSE
+├── mkdocs.yaml
+├── dev-requirements.txt
+└── pyproject.toml
+```
+
+To identify `pynxtools-plugin` as a plugin for pynxtools, an entry point must be established:
+```
+[project.entry-points."pynxtools.reader"]
+mydatareader = "pynxtools_plugin.reader:MyDataReader"
+```
+
+Here, we will focus mostly on the `reader.py` file and how to build a reader. For guidelines on how to build the other parts of your plugin, you can have a look here:
+
+- [Documentation writing guide](https://nomad-lab.eu/prod/v1/staging/docs/writing_guide.html)
+- [Plugin testing framework](using-pynxtools-test-framework.md)
+
+<!-- Note: There is also a [cookiecutter template](https://github.com/FAIRmat-NFDI/pynxtools-plugin-template) available for creating your own pynxtools plugin, but this is currently not well-maintained.-->
+
+
+## Writing a Reader
+
+After you have established the main structure, you can start writing your reader. The new reader shall be placed in `reader.py`.
+
+Then implement the reader function:
+
+```python
+"""MyDataReader implementation for the DataConverter to convert mydata to NeXus."""
+from typing import Tuple, Any
+
+from pynxtools.dataconverter.readers.base.reader import BaseReader
+
+class MyDataReader(BaseReader):
+    """MyDataReader implementation for the DataConverter to convert mydata to NeXus."""
+
+    def read(
+        self,
+        template: dict = None,
+        file_paths: Tuple[str] = None,
+        objects: Tuple[Any] = None
+    ) -> dict:
+        """Reads data from given file and returns a filled template dictionary"""
+        # Fill the template
+        for path in file_paths:
+            print(path)
+
+        template["/entry/instrument/scan"] = raw_scan_data
+
+        return template
+
+
+# This has to be set to allow the convert script to use this reader. Set it to "MyDataReader".
+READER = MyDataReader
+
+
+
+```
+### The reader template dictionary
+
+The read function takes a [`Template`](https://github.com/FAIRmat-NFDI/pynxtools/blob/master/src/pynxtools/dataconverter/template.py) dictionary, which is used to map from the measurement (meta)data to the concepts defined in the application definition. It contains keys based on the provided NXDL file (you can get an empty template by using `dataconverter generate-template`).
```

Comment:
template

Reply:
Instead of "It" make a more specific reference, the appdef contains keys but these are NeXus concept names and the template contains keys NeXus template dictionary keys

Reply:
used uppercase `Template` here because it is a class.

Reply:
adjusted the text